### PR TITLE
Sparse: Use per-operator headers

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #71217
* #71216
* #71215
* #71214
* #71213
* #71212
* #71137
* __->__ #71115
* #71114
* #68462

Differential Revision: [D33949904](https://our.internmc.facebook.com/intern/diff/D33949904)

### CODE REVIEW

File: aten/src/ATen/native/sparse/cuda/SoftMax.cu [hunk 1]
```diff
@@ -13,6 +13,21 @@
...
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/_masked_softmax_native.h>
+#include <ATen/ops/_log_softmax_cuda_dispatch.h>
+#include <ATen/ops/_log_softmax_backward_data_cuda_dispatch.h>
+#include <ATen/ops/_softmax_cuda_dispatch.h>
+#include <ATen/ops/_softmax_backward_data_cuda_dispatch.h>
+#include <ATen/ops/equal_native.h>
+#include <ATen/ops/full.h>
+#include <ATen/ops/softmax.h>
```
Comment (peterbell10) [inline aten/src/ATen/native/sparse/cuda/SoftMax.cu:28]: I used iwyu for all the `.cpp` files which is why that PR is in this stack. Unfortunately iwyu doesn't work with CUDA so the `.cu` files have to be manually looked through to find the required headers.

**Review by IvanYashchuk — APPROVED** [review]
Is there a CI job with `AT_PER_OPERATOR_HEADERS` defined?

**Comment by peterbell10** [comment]
>Is there a CI job with `AT_PER_OPERATOR_HEADERS` defined?

It's the default

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33949904).

**Pull request closed without merging**